docs/debugger_v2.md

```diff
@@ -0,0 +1,303 @@
+constructs, loss function, and training loop, because the likelihood of NaN bugs
+is higher when we use this more flexible but more error-prone API than when we
+use the easier-to-use but slightly less flexible high-level APIs such as
+[tf.keras](https://www.tensorflow.org/guide/keras).
+
+The program prints a test accuracy after each training step. We can see in the
+console that the test accuracy gets stuck at a near-chance level (~0.1) after
+the first step. This is certainly not how the model training is expected to
+behave: we expect the accuracy to gradually approach 1.0 (100%) as the step
+increases.
+
+```
+Accuracy at step 0: 0.216
+Accuracy at step 1: 0.098
+Accuracy at step 2: 0.098
+Accuracy at step 3: 0.098
+...
+```
+
+An educated guess is that this problem is caused by a numerical instability,
+such as NaN or infinity. However, how do we confirm this is really the case and
+how do we find the TensorFlow operation (op) responsible for generating the
+numerical instability? To answer these questions, let’s instrument the buggy
+program with Debugger V2.
+
+## Instrumenting TensorFlow code with Debugger V2
+
+[`tf.debugging.experimental.enable_dump_debug_info()`](https://www.tensorflow.org/api_docs/python/tf/debugging/experimental/enable_dump_debug_info)
+is the API entry point of Debugger V2. It instruments a TF2 program with a
+single line of code. For instance, adding the following line near the beginning
+of the program will cause debug information to be written to the log directory
+(logdir) at /tmp/tfdbg2_logdir. The debug information covers various aspects of
+TensorFlow runtime. In TF2, it includes the full history of eager execution,
+graph building performed by
+[@tf.function](https://www.tensorflow.org/api_docs/python/tf/function),
+the execution of the graphs, the tensor values generated by the execution
+events, as well as the code location (Python stack traces) of those events. The
+richness of the debug information enables users to narrow in on obscure bugs.
+
+```py
+tf.debugging.experimental.enable_dump_debug_info(
+    logdir="/tmp/tfdbg2_logdir",
+    tensor_debug_mode="FULL_HEALTH",
+    circular_buffer_size=-1)
+```
+
+The tensor_debug_mode argument controls what information Debugger V2 extracts
+from each eager or in-graph tensor. “FULL_HEALTH” is a mode that captures the
+following information about each floating-type tensor (e.g., the commonly-seen
+float32 and the less common
+[bfloat16](https://en.wikipedia.org/wiki/Bfloat16_floating-point_format) dtype):
+- DType
+- Rank
+- Total number of elements
+-  A breakdown of the floating-type elements into the following categories:
+   negative finite (`-`), zero (`0`), positive finite (`+`), negative infinity
+   (`-∞`), positive infinity (`+∞`), and `NaN`.
+
+The “FULL_HEALTH” mode is suitable for debugging bugs involving NaN and infinity.
+See below for other supported `tensor_debug_mode`s.
```

Comment:
nit.  consistency.  tensor_debug_mode here is in backtick quotes, where above it is not.

Reply:
Done a pass throughout the doc and using backticks around tensor_debug_mode consistently now.